pkgs/shells/zsh/default.nix

```diff
@@ -42,7 +42,7 @@ stdenv.mkDerivation {
     "--enable-multibyte"
     "--with-tcsetpgrp"
     "--enable-pcre"
-    "--enable-zprofile=${placeholder "out"}/etc/zprofile"
```

Comment:
I still have the gut feeling that this new behavior will break someones use case as in https://xkcd.com/1172/. So better have a changelog in `nixos/doc/manual/release-notes/rl-2305.section.md` for zsh to document the change behavior.